### ISSUE TITLE
Adding docs to separate index_names adds to same record-pool; hence search also looks across index_names

### ISSUE THREAD

**Issue by hschmied** [langchain-ai/langchain-redis#23]
I am creating multiple redisvectorstores on the same redis-db, but different index_names. When I check via redis-cli, the records all seem to land in a 'global-pool' of records, that all indexes access.

To be more specific...
- created 'index_a', 'index_b' & 'index_c' each with 1 record
- when checking 'FT.INFO index_a' it shows 'num_docs: 3'
- when I do a similarity_search with vector_store initialized for 'index_a', I get also back the records from index_b & index_c

Is that a known issue? Am I missing something?

---

before switching to langchain-redis I was using langchain_community.vectorstores.redis as described in langchain v0.1-docs

Thanks!

**Comment by tylerhutcherson** [comment]
We've flagged this -- thanks @hschmied we should have a resolution soon. Thanks for the early feedback/catch.

**Comment by hschmied** [comment]
Actually, I probably should have given it a bit more thought before I posted. I checked the 'latest' docs on how to configure the schema. When I first implemented it (pre v0.3 langchain release) I think I found somewhere a doc that defined the RedisConfig schema via IndexSchema (see below)...
```
            # metadata_schema = default_metadata_schema
            # index_dict = {
            #     "index": { "name": index, "storage_type": "hash" },
            #     "fields": [
            #             {"name": "indexed_metadata", "type": "text"},
            #             {
            #                 "name": "embedding",
            #                 "type": "vector",
            #                 "attrs": {
            #                     "dims": 1536,
            #                     "distance_metric": "cosine",
            #                     "algorithm": "FLAT",
            #                 },
            #             },
            #     ],
            # }
            # index_dict["fields"].extend(metadata_schema)
            # if default_schema is False:
            #     for m in metadata_keys:
            #         if m not in [f["name"] for f in index_dict["fields"]]:
            #             index_dict["fields"].append({ "name": m, "type": "text" })

            # redis_config = RedisConfig(
            #     index_name=index,
            #     redis_url=self.redis_url,
            #     schema=IndexSchema.from_dict(index_dict)
            # )
```

I've now commented that out and followed the latest docs where metadata_schema is used with an array of 'name: type'. Now my tests show 1 record in each index and search also sticks to the given index_name.

Thanks & sorry for the bother.

**Comment by bsbodden** [comment]
@tylerhutcherson anything we should do here to make it the usage clearer?

**Comment by hschmied** [comment]
hmmm... since I came from the 'old' langchain + redis implementation it might have just be a migration-misunderstanding on my part. before there was this setup with vector-dimensions, algorithm and stuff... now there's the setup via config. when I looked into the source, I wasn't sure whether or at what point to use 'schema', 'schema_path', 'existing + index_name' or try to fallback into the else...

![image](https://github.com/user-attachments/assets/0743aa23-c4e1-4b19-ae44-5839dfa1f1a5)

E.g. when providing 'index_schema' how does the 'index_name' i provide actually come into play. things like that

--

even before I migrated, i already struggled a bit with schemas, since 'basic'-imports might have one set of fields, while another has some additional. later that caused problems when trying to run a search and the structure/schema was unclear or mixed. still figuring things out. :)
